### PR TITLE
CORE-6756: Apply feedback from technical writers

### PR DESCRIPTION
**Pull request by josephzunigadaly**
Apply kdoc feedback to serialization classes.

### CODE REVIEW

File: base/src/main/kotlin/net/corda/v5/base/annotations/DeprecatedConstructorForDeserialization.kt [hunk 1]
```diff
@@ -6,6 +6,8 @@ package net.corda.v5.base.annotations
  *
  * Versions will be considered in descending order, currently duplicate versions will result in
  * non-deterministic behaviour when deserializing objects
+ *
+ * @property version integer defining order that these annotations will be applied
```
Comment (nadinequinr3) [inline base/src/main/kotlin/net/corda/v5/base/annotations/DeprecatedConstructorForDeserialization.kt]:
```suggestion
 * @property version Integer defining order that these annotations will be applied.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 2]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
         this.typeNames = List.copyOf(typeNames);
     }
 
+    /**
+     * Constructs a MissingSerializerException with the specified message and type descriptor
+     * @param message error message describing exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @param message Error message describing exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 3]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     * @param typeDescriptor type descriptor to help diagnose the exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @param typeDescriptor Type descriptor to help diagnose the exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 4]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     */
     public MissingSerializerException(@NotNull String message, @NotNull String typeDescriptor) {
         this(message, typeDescriptor, emptyList());
         requireNonNull(typeDescriptor, "typeDescriptor may not be null");
     }
 
+    /**
+     * Constructs a MissingSerializerException with the specified message and type names
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * Constructs a MissingSerializerException with the specified message and type names.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 5]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     * @param message error message describing exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @param message Error message describing the exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 6]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     * @param typeNames type names to help diagnose the exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @param typeNames Type names to help diagnose the exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 7]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     */
     public MissingSerializerException(@NotNull String message, @NotNull List<String> typeNames) {
         this(message, null, typeNames);
     }
 
+    /**
+     * Returns the type descriptor stored in this exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]: Is this needed with the line below?

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 8]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     * @return the type descriptor stored in this exception
+     */
     @Nullable
     public String getTypeDescriptor() {
         return typeDescriptor;
     }
 
+    /**
+     * Returns the type names stored in this exception
+     * @return the type names stored in this exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @return The type names stored in this exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java [hunk 9]
```diff
@@ -3,17 +3,17 @@
...
  * unserializable types and an intermediate representation.
  *
  * NOTE: The proxy object should be specified as a separate class. However, this can be defined within the
  * scope of the custom serializer.
  */
 public interface SerializationCustomSerializer<OBJ, PROXY> {
     /**
-     * Should facilitate the conversion of the third party object into the serializable
+     * Facilitates the conversion of the third party object into the serializable
      * local class specified by {@code PROXY}
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java]:
```suggestion
     * local class specified by {@code PROXY}.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java [hunk 10]
```diff
@@ -3,17 +3,17 @@
...
      * @param obj original object for serialization
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java]:
```suggestion
     * @param obj Original object for serialization.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java [hunk 11]
```diff
@@ -22,7 +22,7 @@
     PROXY toProxy(@NotNull OBJ obj);
 
     /**
-     * Should facilitate the conversion of the proxy object into a new instance of the
+     * Facilitates the conversion of the proxy object into a new instance of the
      * unserializable type
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java]:
```suggestion
     * unserializable type.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java [hunk 12]
```diff
@@ -22,7 +22,7 @@
...
      * @param proxy object from AMQP
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializationCustomSerializer.java]:
```suggestion
     * @param proxy Object from AMQP.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java [hunk 13]
```diff
@@ -10,10 +10,18 @@
 @SuppressWarnings("unused")
 @CordaSerializable
 public final class SerializedBytes<T> extends OpaqueBytes {
+    /**
+     * Constructs a SerializedBytes holding the specified bytes
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java]:
```suggestion
     * Constructs a SerializedBytes holding the specified bytes.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java [hunk 14]
```diff
@@ -10,10 +10,18 @@
...
+     * @param bytes byte array holding serialized form of type T
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java]:
```suggestion
     * @param bytes Byte array holding serialized form of type T.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java [hunk 15]
```diff
@@ -10,10 +10,18 @@
...
+     */
     public SerializedBytes(@NotNull byte[] bytes) {
         super(bytes);
     }
 
+    /**
+     * Returns a text description of this SerializedBytes object
+     * @return description of this SerializedBytes object
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java]:
```suggestion
     * @return Text description of this SerializedBytes object.
```

File: serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java [hunk 16]
```diff
@@ -10,10 +10,18 @@
...
+    /**
+     * Constructs a SerializedBytes holding the specified bytes
+     * @param bytes byte array holding serialized form of type T
+     */
     public SerializedBytes(@NotNull byte[] bytes) {
         super(bytes);
     }
 
+    /**
+     * Returns a text description of this SerializedBytes object
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java:22]: needed?

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java (hunk 7 above)
Comment (josephzunigadaly) [reply serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]: It represents different fields within the documentation system. The first line is the method description the second line is the description of the return value. In this case the method is so simple it appears to be duplicate content.

I think it's better to have duplicate content than have a missing method description.

File: serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java (hunk 16 above)
Comment (josephzunigadaly) [reply serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java:22]: yes, it's the method description/return value description again

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java (hunk 7 above)
Comment (nadinequinr3) [reply serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]: Definitely, my bad!

Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * Returns the type descriptor stored in this exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 17]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     * @param message error message describing exception
+     * @param typeNames type names to help diagnose the exception
+     */
     public MissingSerializerException(@NotNull String message, @NotNull List<String> typeNames) {
         this(message, null, typeNames);
     }
 
+    /**
+     * Returns the type descriptor stored in this exception
+     * @return the type descriptor stored in this exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @return The type descriptor stored in this exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java [hunk 18]
```diff
@@ -28,20 +28,38 @@ private MissingSerializerException(
...
+     */
     @Nullable
     public String getTypeDescriptor() {
         return typeDescriptor;
     }
 
+    /**
+     * Returns the type names stored in this exception
```
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * Returns the type names stored in this exception.
```

File: serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java (hunk 8 above)
Comment (nadinequinr3) [inline serialization/src/main/java/net/corda/v5/serialization/MissingSerializerException.java]:
```suggestion
     * @return The type names stored in this exception.
```

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

**Review by nadinequinr3 — APPROVED** [review]

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**